src/audio/kpb.c

```diff
@@ -1388,26 +1391,56 @@ static void kpb_reset_history_buffer(struct hb *buff)
 	} while (buff != first_buff);
 }
 
-static inline bool validate_host_params(size_t host_period_size,
-					size_t host_buffer_size,
-					size_t bytes_per_ms)
+static inline bool validate_host_params(struct comp_dev *dev,
+					size_t host_period_size,
+					size_t host_buffer_size)
 {
-	/* Check host period size sanity.
-	 * Here we check if host period size (which defines interval
-	 * time) will allow us to drain more data then the interval
-	 * takes - as only such condition guarantees draining will end.
-	 * The formula:
-	 *	drained_data_in_one_interval_ms > interval_break_ms
-	 * where:
-	 * drained_data_in_one_interval_ms = (host_period_size * 2) [ms]
-	 * interval_break_ms = host_period_size / bytes_per_ms [ms]
+	/* The aim of this function is to perform basic check of host params
+	 * and reject them if they won't allow for stable draining.
+	 * Note however that this is highly recommended for host buffer to
+	 * be of history buffer size. This will quarantee "safe" draining.
+	 * By safe we mean no XRUNs(host was unable to read data on time),
+	 * or lose of data due to host delayed read. The later condition
+	 * is very likely after wake up from power state like d0ix.
 	 */
+	struct comp_data *kpb = comp_get_drvdata(dev);
+	size_t sample_width = kpb->config.sampling_width;
+	size_t bytes_per_ms = KPB_SAMPLING_WIDTH *
+			      (KPB_SAMPLE_CONTAINER_SIZE(sample_width) / 8) *
+			      kpb->config.no_channels;
```

Comment:
this isn't introduced by this patch, but this seems a bit confusing to have `.no_channels` to mean "number of channels?"  I think `num_` or `n_` are usually used

Reply:
Yes I agree with you and I will refactor it adding another commit.